agents/aliyun/fence_aliyun.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/python -tt
+
+import sys, re
+import logging
+import atexit
+import json
+sys.path.append("@FENCEAGENTSLIBDIR@")
+from fencing import *
+from fencing import fail, fail_usage, EC_TIMED_OUT, run_delay
+
+from aliyunsdkcore import client
```

Comment:
You'll have to add try/except here so it can generate metadata without the libraries:
try:
    import ...
except ImportError: pass